sdks/java/io/amazon-web-services/src/main/java/org/apache/beam/sdk/io/aws/sns/SnsIO.java

```diff
@@ -256,9 +269,36 @@ public Write withResultOutputTag(TupleTag<PublishResult> results) {
       return builder().setResultOutputTag(results).build();
     }
 
+    /**
+     * Encode the full {@code PublishResult} object, including HTTP status code and response
+     * headers.
+     */
+    public Write withFullPublishResult() {
+      return builder()
+          .setCoderProvider(
+              CoderProviders.forCoder(
+                  TypeDescriptor.of(PublishResult.class), FullPublishResultCoder.of()))
+          .build();
+    }
+
+    /**
+     * Encode the minimal {@code PublishResult} object, including HTTP status code but excluding
+     * HTTP response headers.
+     */
+    public Write withMinimalPublishResult() {
+      return builder()
+          .setCoderProvider(
+              CoderProviders.forCoder(
+                  TypeDescriptor.of(PublishResult.class), MinimalPublishResultCoder.of()))
+          .build();
+    }
+
     @Override
     public PCollectionTuple expand(PCollection<PublishRequest> input) {
       checkArgument(getTopicName() != null, "withTopicName() is required");
+      if (getCoderProvider() != null) {
```

Comment:
I don't think we want to use a coder provider here but just a coder directly and call `setCoder` on the resulting `PCollection` from `input.apply`

Coder providers are meant to allow users a way to bind object -> coder resolution logic for an entire pipeline (not just a single transform).

Reply:
I understand now, thanks.